src/core/appgateway.tf

```diff
@@ -456,28 +456,64 @@ module "app_gw" {
   rewrite_rule_sets = [
     {
       name = "rewrite-rule-set-api"
-      rewrite_rules = [{
-        name          = "http-headers-api"
-        rule_sequence = 100
-        conditions    = []
-        url           = null
-        request_header_configurations = [
-          {
-            header_name  = "X-Forwarded-For"
-            header_value = "{var_client_ip}"
-          },
-          {
-            header_name  = "X-Client-Ip"
-            header_value = "{var_client_ip}"
-          },
-          {
-            # this header will be checked in apim policy (only for MTLS check)
-            header_name  = data.azurerm_key_vault_secret.app_gw_mtls_header_name.value
-            header_value = "false"
-          },
-        ]
-        response_header_configurations = []
-      }]
+      rewrite_rules = [
+        {
+          name          = "http-headers-api"
+          rule_sequence = 100
+          conditions    = []
+          url           = null
+          request_header_configurations = [
+            {
+              header_name  = "X-Forwarded-For"
+              header_value = "{var_client_ip}"
+            },
+            {
+              header_name  = "X-Client-Ip"
+              header_value = "{var_client_ip}"
+            },
+            {
+              # this header will be checked in apim policy (only for MTLS check)
+              header_name  = data.azurerm_key_vault_secret.app_gw_mtls_header_name.value
+              header_value = "false"
+            },
+          ]
+          response_header_configurations = []
+        },
+        {
+          name          = "url-rewrite-openid-provider-private"
+          rule_sequence = 200
+          conditions = [
+            {
+              ignore_case = true
+              pattern     = "\\/openid-provider\\/admin\\/(.*)"
```

Comment:
please use variables (a variable with a list of blocked paths)

Reply:
Does it make sense to support multiple patterns when we need a single one now? Could be the case to update the code when we actually need for it?